OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java

```diff
@@ -531,6 +531,67 @@ public int[] getTypes() {
 		return types;
 	}
 
+	public static final String RULE_INT_MAX = "RULE_INT_MAX";
+
+	public void applyBoostedConditionalTags(Map<String, String> boostConditionalTags) {
```

Comment:
1. Unrelated to RouteDataObject class
2. No check that object ahs any condtional tag and all these cycles are needed